tests/test_flagging_clustering.py

```diff
@@ -0,0 +1,74 @@
+"""Tests for clustering and flagging"""
+import logging
+import os
+import unittest
+import pytest
+import copy
+
+import astropy.units as u
+import numpy as np
+
+import pint.models.model_builder as mb
+import pint.toa as toa
+from pint.residuals import Residuals
+from pinttestdata import datadir
+from pint.models import parameter as p
+from pint.models import PhaseJump
+
+
+class SimpleSetup:
+    def __init__(self, par, tim):
+        self.par = par
+        self.tim = tim
+        self.m = mb.get_model(self.par)
+        self.t = toa.get_TOAs(
+            self.tim, ephem="DE405", planets=False, include_bipm=False
+        )
+
+
+@pytest.fixture
+def setup_NGC6440E():
+    os.chdir(datadir)
```

Comment:
Do not change directory in tests, just use `os.path.join(datadir, filename)`; changing directories changes global state.

Reply:
sure, although I was just copying some other tests.

Reply:
Yes, we have a lot of bad tests. I'm trying to avoid adding new bad ones.

Reply:
Sure, I'm not objecting.  I'm just pointing out that a more systematic change to tests could be worthwhile.

Reply:
I thought about this when I came back to active work on PINT, but the large number of tests and the difficulty of understanding what they are testing makes it an extremely daunting task to even convert them to a modern style, let alone to understandable unit, integration, or comparison tests. So I have been following the Scouts' rule of leaving everywhere better than I found it - when I have to modify some code I clean it up along the way.